### PR TITLE
Moving fastforward from cmd to pkg

### PR DESCRIPTION
**Pull request by rajibmitra**
#### What this PR does / why we need it:
we are in process of moving fastforward from cmd to pkg. 
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/fastforward/fastforward.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kgit "k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/util"
+)
+
+//  FFrootOptions is the main structure for fastforward.
+type FFrootOptions struct {
```
Comment (saschagrunert) [inline pkg/fastforward/fastforward.go:29]:
```suggestion
type Options struct {
```

File: pkg/fastforward/fastforward.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	Branch         string
+	MainRef        string
+	NonInteractive bool
+	Nomock         bool
+	Cleanup        bool
+	RepoPath       string
+	LogLevel       string
+}
+
+var Options = &FFrootOptions{}
```
Comment (saschagrunert) [inline pkg/fastforward/fastforward.go:39]: I think we don't need this:
```suggestion
```

File: pkg/fastforward/fastforward.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+const pushUpstreamQuestion = `Are you ready to push the local branch fast-forward changes upstream?
+Please only answer after you have validated the changes.`
+
+func RunFf(opts *FFrootOptions) error {
```
Comment (saschagrunert) [inline pkg/fastforward/fastforward.go:44]:
```suggestion
func Run(opts *Options) error {
```

File: cmd/krel/cmd/ff.go [hunk 4]
```diff
@@ -37,7 +35,13 @@ type ffOptions struct {
 	cleanup        bool
 }
 
-var ffOpts = &ffOptions{}
+var ffOpts = &ffOptions{
```
Comment (saschagrunert) [inline cmd/krel/cmd/ff.go:38]:
```suggestion
var ffOpts = &fastforward.Options{}
```

File: cmd/krel/cmd/ff.go [hunk 5]
```diff
@@ -62,13 +66,10 @@ as real push if the '--nomock' flag is specified.
 	SilenceUsage:  true,
 	SilenceErrors: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return runFf(ffOpts, rootOpts)
+		return fastforward.RunFf(fastforward.Options)
```
Comment (saschagrunert) [inline cmd/krel/cmd/ff.go:69]: We have to set the `nomock` here, then pass the `ffOpts`.
```suggestion
		ffOpts.NoMock = rootOpts.nomock
		return fastforward.RunFf(ffOpts)
```

File: pkg/fastforward/fastforward.go [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+	kgit "k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/util"
+)
+
+//  FFrootOptions is the main structure for fastforward.
+type FFrootOptions struct {
+	Branch         string
+	MainRef        string
+	NonInteractive bool
+	Nomock         bool
```
Comment (saschagrunert) [inline pkg/fastforward/fastforward.go:33]:
```suggestion
	NoMock         bool
```

**Comment by rajibmitra** [comment]
thanks @saschagrunert  for the the suggestions, got it now :)

**Comment by saschagrunert** [comment]
> thanks @saschagrunert for the the suggestions, got it now :)

There are some small fixups neccesary, see CI build :) Can you also squash your commits together please?

File: pkg/fastforward/fastforward.go [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kgit "k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/util"
+)
+
+//  FFrootOptions is the main structure for fastforward.
```
Comment (saschagrunert) [inline pkg/fastforward/fastforward.go:28]:
```suggestion
//  Options is the main structure for configuring a fast forward.
```

File: cmd/krel/cmd/ff_test.go [hunk 8]
```diff
@@ -120,13 +120,13 @@ func TestFfSuccess(t *testing.T) {
 	defer s.cleanup(t)
 
 	ffo := s.getFfOptions()
-	ffo.branch = pseudoReleaseBranch
+	ffo.Branch = pseudoReleaseBranch
 
 	ro := s.getRootOptions()
 	ro.nomock = true
```
Comment (saschagrunert) [inline cmd/krel/cmd/ff_test.go:126]: We do not need the root options any more, but we have to set the nomock for the ffo:
```suggestion
	ro := s.getRootOptions()
	ffo.NoMock = true
```

**Comment by rajibmitra** [comment]
I will do rebase once ci looks good. Thanks @saschagrunert for helping me with this. :+1:

**Comment by rajibmitra** [comment]
rebased with squash.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**